jsystemd-spring-boot-starter/src/main/java/com/github/jpmsilva/jsystemd/SystemdNotifyApplicationContextStatusProvider.java

```diff
@@ -79,7 +73,7 @@ public Object postProcessBeforeInitialization(Object bean, String beanName)
       throws BeansException {
     if (factory != null) {
       ensureDefinitionsLoaded();
-      status = String.format("Creating bean %d of %d", getSingletonCount(), definitions.size());
+      status = String.format("Application %d (%s): creating bean %d of %d", applicationId, contextId, getSingletonCount(), definitions.size());
```

Comment:
Can you clarify if the two application contexts are started concurrently?

If they do, I would see this as somewhat problematic, given that the status would "jump" continuously between application contexts, confusing users as to the real progress of the startup.

If they don't, I still this as somewhat problematic, since there would still be a fuzzier representation regarding the actual startup progress. Something like a cumulative progress across all application contexts would be preferable. That being said, I don't really have much at stake here, since I only really work with single application context applications.

Reply:
Yes, the two application contexts are started in parallel but the status will not "jump" between application contexts. The way I implemented it is that the status contains the state of both the application contexts. This is an example of the status line of an application start-up:
Application 127791068 state: starting
Application 1312963234 state: starting, Application 127791068 state: starting
Application 1312963234 state: environment prepared, Application 127791068 state: starting
Application 1312963234 state: context prepared, Application 127791068 state: starting
Application 1312963234 state: context loaded, Application 127791068 state: starting
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 12 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 13 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 14 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 15 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 16 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 17 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 18 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 19 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 20 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 21 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 22 of 22
Application 1312963234 state: context loaded, Application 127791068 state: starting, Application 1312963234 (bootstrap): creating bean 22 of 22
Application 1312963234 state: context loaded, Application 127791068 state: environment prepared, Application 1312963234 (bootstrap): creating bean 22 of 22
Application 1312963234 state: context loaded, Application 127791068 state: context prepared, Application 1312963234 (bootstrap): creating bean 22 of 22
Application 1312963234 state: context loaded, Application 127791068 state: context loaded, Application 1312963234 (bootstrap): creating bean 22 of 22
Application 1312963234 state: context loaded, Application 127791068 state: context loaded, Application 1312963234 (bootstrap): creating bean 22 of 22, Application 127791068 (switch:default:8080): creating bean 36 of 709
Application 1312963234 state: context loaded, Application 127791068 state: context loaded, Application 1312963234 (bootstrap): creating bean 22 of 22, Application 127791068 (switch:default:8080): creating bean 37 of 709
...

Reply:
Ok, that's a nice readable output, I think we can proceed with it.